submissions/f3d0t/friends-app/app.js

```diff
@@ -0,0 +1,173 @@
+			}
+			const checkedUserAge = FILTERS.querySelector("[name=userAge]:checked");
+			if (checkedUserAge) {
+				friends.sortByAge(checkedUserAge.id);
+			} //this two forEach callbacks is needed to re-sort new arrays of cards, filtered by gender
+		}
+		if (target.name === "userName") {
+			friends.sortByName(target.id);
+			FILTERS.elements.search.value = "";
+			const checkedUserAge = FILTERS.querySelector("[name=userAge]:checked");
+			if (checkedUserAge) {
+				checkedUserAge.checked = false;
+			}
+		}
+		if (target.name === "userAge") {
+			friends.sortByAge(target.id);
+			FILTERS.elements.search.value = "";
+		}
+	});
+	RESET_BUTTON.addEventListener("click", (e) => {
+		e.preventDefault();
+		FILTERS.querySelectorAll("input[type=radio]:checked").forEach((radioButton) => {
+			radioButton.checked = false;
+		});
+		FILTERS.elements.search.value = "";
+		friends.resetFilters();
+	});
+	FILTERS.addEventListener("submit", (event) => {
+		if (event.key === "Enter") {
+			event.preventDefault();
+		}
+	});
+};
+
+class FriendsList {
+	constructor(data, parentWrapper) {
+		this.allCards = data.map((friendData) => new FriendCard(friendData));
+		this.currentCards = [...this.allCards];
+		this.wrapper = parentWrapper;
+	}
+	renderCards(cardsArray = this.currentCards) {
+		this.wrapper.innerHTML = "";
+		this.wrapper.append(...cardsArray.map((card) => card.createCard()));
+	}
+	resetFilters() {
+		this.currentCards = [...this.allCards];
+		this.renderCards();
+	}
+	filterByGender(gender) {
+		if (gender === "male") {
+			this.currentCards = this.allCards.filter((friendCard) => friendCard.gender === "male");
+		}
+		if (gender === "female") {
+			this.currentCards = this.allCards.filter((friendCard) => friendCard.gender === "female");
+		}
+		if (gender === "all") {
+			this.currentCards = [...this.allCards];
+		}
+		this.renderCards();
+	}
+	filterBySearch(searchValue) {
+		const filteredArray = this.currentCards.filter((friendCard) =>
+			friendCard.searchTarget.toLowerCase().includes(searchValue.toLowerCase())
+		);
+		this.renderCards(filteredArray);
+	}
+	sortByName(sortType) {
+		if (sortType === "az") {
+			this.currentCards = sortObjectsByPropertyValue(this.currentCards, "name", "ascending");
+		}
+		if (sortType === "za") {
+			this.currentCards = sortObjectsByPropertyValue(this.currentCards, "name", "descending");
+		}
+		this.renderCards();
+	}
+	sortByAge(sortType) {
+		if (sortType === "ascending") {
+			this.currentCards = sortObjectsByPropertyValue(this.currentCards, "age", "ascending");
+		}
+		if (sortType === "descending") {
+			this.currentCards = sortObjectsByPropertyValue(this.currentCards, "age", "descending");
+		}
+		this.renderCards();
+	}
+}
+
+const sortObjectsByPropertyValue = (arrayOfObjects, key, sortOrder) => {
+	const array = arrayOfObjects;
+	array.sort((a, b) => {
+		if (a[key] > b[key]) {
+			return 1;
+		} else if (a[key] < b[key]) {
+			return -1;
+		} else {
+			return 0;
+		}
```

Comment:
Looks cleaner, IMO :)

```suggestion
		if (a[key] > b[key]) {
			return 1;
		} 
		
		if (a[key] < b[key]) {
			return -1;
		}
		
		return 0;
```